### PR TITLE
Dockerize

### PR DESCRIPTION
**Pull request by gurinderu**
Fixed broken musl build
Add docker publishing

### CODE REVIEW

**Comment by gurinderu** [comment]
@poslegm To fix this build you need to add two envs
DOCKER_USERNAME username with permissions for scalameta/scalafmt
DOCKER_PASSWORD access token (you can create a new one [here](https://hub.docker.com/settings/security))

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -68,11 +71,26 @@ jobs:
...
           tar xvf musl.tar.gz
-
       - run: bin/build-native-image.sh
         env:
           CI: true
       - uses: actions/upload-artifact@master
         with:
           name: ${{ matrix.artifact }}
           path: scalafmt
+  dockerize:
```
Comment (poslegm) [inline .github/workflows/ci.yml:81]: Will it push new image on every PR?

**Comment by poslegm** [comment]
@gurinderu as far as I understand secrets for Docker already [added](https://github.com/scalameta/scalafmt/pull/1831#issuecomment-606609018).

Anyway I have not access to repo's secrets and need to summon @olafurpg here

**Comment by olafurpg** [comment]
The docker secrets have already been configured but they're only available after merge into master

File: .github/workflows/ci.yml (hunk 1 above)
Comment (gurinderu) [reply .github/workflows/ci.yml:81]: yep, with a custom tag (refs/pull/2/merge | pr-2-merge)

Comment (poslegm) [reply .github/workflows/ci.yml:81]: Can you please configure it as dedicated workflow running only on tags push? 

```
on:
  push:
    branches: [master]
    tags: ["*"]
```

Image creation on every PR would be redundant

Comment (gurinderu) [reply .github/workflows/ci.yml:81]: Maybe it is better to add this flow only for tags?

Comment (gurinderu) [reply .github/workflows/ci.yml:81]: I added condition for tags only.

Comment (poslegm) [reply .github/workflows/ci.yml:81]: Thank you! I will test it on next release

**Comment by gurinderu** [comment]
@poslegm hey, How did you release v2.6.2 ? I saw the tag, but didn't see any related actions

**Pull request merged**